app/src/test/java/org/oppia/android/app/parser/StringToFractionParserTest.kt

```diff
@@ -0,0 +1,462 @@
+    assertThat(error).isEqualTo(StringToFractionParser.FractionParsingError.VALID)
+  }
+
+  @Test
+  fun testParser_realTimeError_mixedNegativeNumber_returnsValid() {
+    val error = stringToFractionParser.getRealTimeAnswerError("-5 2/3")
+    assertThat(error).isEqualTo(StringToFractionParser.FractionParsingError.VALID)
+  }
+
+  @Test
+  fun testParser_realTimeError_validRegularFraction_noErrorMessage() {
+    val errorMessage = stringToFractionParser.getRealTimeAnswerError("2/3")
+      .getErrorMessageFromStringRes(context)
+    assertThat(errorMessage).isNull()
+  }
+
+  @Test
+  fun testParser_realTimeError_nonDigits_returnsInvalidChars() {
+    val error = stringToFractionParser.getRealTimeAnswerError("abc")
+    assertThat(error).isEqualTo(StringToFractionParser.FractionParsingError.INVALID_CHARS)
+  }
+
+  @Test
+  fun testParser_realTimeError_nonDigits_invalidChars_hasRelevantErrorMessage() {
+    val errorMessage = stringToFractionParser.getRealTimeAnswerError("abc")
+      .getErrorMessageFromStringRes(context)
+    assertThat(errorMessage)
+      .isEqualTo("Please only use numerical digits, spaces or forward slashes (/)")
+  }
+
+  @Test
+  fun testParser_realTimeError_noNumerator_returnsInvalidFormat() {
+    val error = stringToFractionParser.getRealTimeAnswerError("/3")
+    assertThat(error).isEqualTo(StringToFractionParser.FractionParsingError.INVALID_FORMAT)
+  }
+
+  @Test
+  fun testParser_realTimeError_noNumerator_invalidFormat_hasRelevantErrorMessage() {
+    val errorMessage = stringToFractionParser.getRealTimeAnswerError("/3")
+      .getErrorMessageFromStringRes(context)
+    assertThat(errorMessage)
+      .isEqualTo("Please enter a valid fraction (e.g., 5/3 or 1 2/3)")
+  }
+
+  @Test
+  fun testParser_realTimeError_severalSlashes_invalidFormat_returnsInvalidFormat() {
+    val error = stringToFractionParser.getRealTimeAnswerError("1/3/8")
+    assertThat(error).isEqualTo(StringToFractionParser.FractionParsingError.INVALID_FORMAT)
+  }
+
+  @Test
+  fun testParser_realTimeError_severalSlashes_invalidFormat_hasRelevantErrorMessage() {
+    val errorMessage = stringToFractionParser.getRealTimeAnswerError("1/3/8")
+      .getErrorMessageFromStringRes(context)
+    assertThat(errorMessage)
+      .isEqualTo("Please enter a valid fraction (e.g., 5/3 or 1 2/3)")
+  }
+
+  @Test
+  fun testParser_realTimeError_severalDashes_returnsInvalidFormat() {
+    val error = stringToFractionParser.getRealTimeAnswerError("-1/-3")
+    assertThat(error).isEqualTo(StringToFractionParser.FractionParsingError.INVALID_FORMAT)
+  }
+
+  @Test
+  fun testParser_realTimeError_severalDashes_invalidFormat_hasRelevantErrorMessage() {
+    val errorMessage = stringToFractionParser.getRealTimeAnswerError("-1/-3")
+      .getErrorMessageFromStringRes(context)
+    assertThat(errorMessage)
+      .isEqualTo("Please enter a valid fraction (e.g., 5/3 or 1 2/3)")
+  }
+
+  @Test
+  fun testParser_parseFraction_divisionByZero_parsesFraction() {
```

Comment:
Here & below: I suggest also keeping clear condition/outcomes. E.g.:

- ``testParseFraction_divisionByZero_returnsFraction``
- ``testParseFraction_multipleFractions_failsWithError``
- ``testParseFraction_regularFraction_returnsFraction``

etc.

Reply:
Done. Renamed here, below and some above.